be/src/formats/parquet/level_builder.cpp

```diff
@@ -0,0 +1,656 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "formats/parquet/level_builder.h"
+
+#include <parquet/arrow/writer.h>
+
+#include <utility>
+
+#include "column/array_column.h"
+#include "column/column_helper.h"
+#include "column/map_column.h"
+#include "column/struct_column.h"
+#include "column/vectorized_fwd.h"
+#include "common/logging.h"
+#include "exprs/expr.h"
+#include "gutil/casts.h"
+#include "gutil/endian.h"
+#include "util/defer_op.h"
+
+namespace starrocks::parquet {
+
+inline uint8_t* get_raw_null_column(const ColumnPtr& col) {
+    if (!col->has_null()) {
+        return nullptr;
+    }
+    auto null_column = down_cast<NullableColumn*>(col.get())->null_column();
+    auto raw_column = null_column->get_data().data();
+    return raw_column;
+}
+
+template <LogicalType lt>
+inline RunTimeCppType<lt>* get_raw_data_column(const ColumnPtr& col) {
```

Comment:
duplicate code, refer to `ColumnHelper::cast_to_raw`

Reply:
It's a combination of `ColumnHelper::get_data_column` and `ColumnHelper::cast_to_raw`